testing/cli/test_cli_sct_crop_image.py

```diff
@@ -13,6 +13,7 @@
     ('t2/t2.nii.gz', 't2_crop_xyz.nii', ['-xmin', '1', '-xmax', '-3', '-ymin', '2', '-ymax', '10'], (57, 9, 52)),
     ('t2/t2.nii.gz', 't2_crop_mask.nii', ['-m', 't2/t2_seg-manual.nii.gz'], (11, 55, 13)),
     ('t2/t2.nii.gz', 't2_crop_ref.nii', ['-ref', 'mt/mt0.nii.gz'], (37, 55, 34)),
+    ('qform/bad_quaternion.nii.gz', 'bad_quaternion_crop.nii', [], (2, 3, 4)),
```

Comment:
I feel a bit hesitant about adding this image to `sct_testing_data`:

- It's a little opaque to have to read the [`README.md`](https://github.com/spinalcordtoolbox/sct_testing_data/pull/10/files#diff-a3ac5c739d67ad8ef086f9ebcec1ebbbac57dd64a47823561e969a3985f2e760) to find out what `bad_quaternion.nii.gz` is actually doing.
- A new image might be overkill considering the only condition we're testing is a slightly off quarternion.
- `sct_testing_data` already feels quite bloated with niche, corner-case images (not the end of the world, but...)

So, my thinking here is that it might be better to take the existing file `t2.nii.gz` and tweak the quarternion on the fly, then save that modified image to a temporary directory. (That way, we can see exactly what it is we're testing inside the test itself, and we don't need to add a new image to `sct_testing_data`.)

Something like:

```python
@pytest.fixture()
def bad_quaternion(input_file, tmp_path):
    """This function modifies the quaternion of the image so that the 'quatern' values (b, c, d)
    no longer add to 1."""
	im_in = Image(input_file)
    im_in.header['quatern_d'] += 0.00097656
    fname_out = tmp_path / 'bad_quaternion.nii.gz'
    im_in.save(fname_out)
    return fname_out
```

However, this runs into a completely separate issue: pytest [doesn't allow you to use fixtures in `pytest.mark.parametrize`'d tests](https://github.com/pytest-dev/pytest/issues/349). 

This is... the one glaring fault I have with Pytest, and it's a bit awkward to work around, but... maybe we could just... create a separate test for this, instead of re-using the parametrized test?

Reply:
Thanks for your thoughts. Here are some of mine, with no particular conclusion yet.
* I'd be happy to not add this image to `sct_testing_data`, if only because it seems like a heavy process to create a new release of another repository to add a test case in this repository.
* Relatedly, I was a bit surprised to see that we don't have anywhere in this repository to have small test data files? Could we consider adding a `testing/data/` directory beside `testing/api/` and `testing/cli/`?
* Alternatively, we could take the stance that all of our testing data should be programmatically generated. But then silent bugs in the generation code would mean that we're not really testing what we _think_ we're testing. For example, what if the image saving code automatically normalizes the quaternions? Or what if we're trying to test behaviour on ill-formed data files?
* (A nitpick: I know the example `bad_quaternion` fixture is only for illustrative purposes, but I'll note that it's the sum of _squares_ of quaternions that we're concerned with, and that typically the sum is _at most_ 1, so it would be more robust to overwrite all three quaternions with known values.)
* Regarding the corner-case-ness of `sct_testing_data`, I would have assumed that that was the point, especially since we also have `sct_example_data` for more realistic or typical data. In fact I would argue that corner cases provide a lot of value for automated testing, since they guard against regressions.

Reply:
Thank you for your thoughts!! I very much agree with all of what you said. 

To be completely honest, testing data is... something that I have a lot of half-formed thoughts about? I agree that it's not exactly clear when SCT devs should use `sct_example_data`, `sct_testing_data`, `sct_tutorial_data`, a programmatically-generated dummy file, or an in-repo file. 

This has been the source of a few different ideas and issues (https://github.com/spinalcordtoolbox/spinalcordtoolbox/issues/3471, https://github.com/spinalcordtoolbox/spinalcordtoolbox/issues/3136, and tangentially related https://github.com/spinalcordtoolbox/spinalcordtoolbox/issues/3508), but, I haven't really sat down to settle on a clear set of recommendations. 

The closest I got was the ["Testing: Datasets"](https://github.com/spinalcordtoolbox/spinalcordtoolbox/wiki/Testing%3A-Datasets) wiki page, but even that is... not clear enough in its recommendations, I think.

Reply:
In the short-term, could we use a programmatically generated "bad quaternion" file, but then following the 5.6 release, sit down to properly outline how we want to approach testing data when working together?